unitTests/ui/QueryForCommerceTest.ts

```diff
@@ -0,0 +1,82 @@
+import { QueryForCommerce, IQueryForCommerceOptions } from '../../src/ui/QueryForCommerce/QueryForCommerce';
+import * as Mock from '../MockEnvironment';
+import { Simulate } from '../Simulate';
+import { AnalyticsEvents, $$ } from '../../src/Core';
+import { IChangeableAnalyticsDataObject, IChangeAnalyticsCustomDataEventArgs } from '../../src/events/AnalyticsEvents';
+import { analyticsActionCauseList } from '../../src/ui/Analytics/AnalyticsActionListMeta';
+
+export function QueryForCommerceTest() {
+  describe('QueryForCommerce', function() {
+    let test: Mock.IBasicComponentSetup<QueryForCommerce>;
+
+    const originalAnalyticsDataObject: IChangeableAnalyticsDataObject = {
```

Comment:
I think I would move that object to a `describe` block that is for all the UA tests, instead of globally for all of the tests of the component.